### ISSUE TITLE
modelEstimation example code produces empty columns in Daily data frame

### ISSUE THREAD

**Issue by jsta** [USGS-R/EGRET#305]
I feel like I _must_ be doing something wrong. Hopefully someone else can reproduce this error. 

Before running `modelEstimation` the example data has entries for the `yHat` column but after running the function this column and all other subsequent columns are completely `NA`. 

```r
library(EGRET)
eList <- Choptank_eList
any(!is.na(eList$Daily$yHat))
# TRUE

eList <- modelEstimation(eList)
any(!is.na(eList$Daily$yHat))
# FALSE
```

<details><summary>session info</summary>

```
setting  value                       
 version  R version 4.0.4 (2021-02-15)
 os       Ubuntu 20.04.2 LTS          
 system   x86_64, linux-gnu           
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/New_York            
 date     2021-03-10 
```

```
package     * version    date       lib source                              
 assertthat    0.2.1      2019-03-21 [1] CRAN (R 4.0.0)                      
 cachem        1.0.3      2021-02-04 [1] CRAN (R 4.0.3)                      
 callr         3.5.1      2020-10-13 [1] CRAN (R 4.0.3)                      
 cli           2.3.1      2021-02-23 [1] CRAN (R 4.0.4)                      
 crayon        1.4.1      2021-02-08 [1] CRAN (R 4.0.3)                      
 datapasta   * 3.1.0      2020-01-17 [1] CRAN (R 4.0.0)                      
 DBI           1.1.1      2021-01-15 [1] CRAN (R 4.0.3)                      
 debugme       1.1.0      2017-10-22 [1] CRAN (R 4.0.3)                      
 desc          1.2.0      2018-05-01 [1] CRAN (R 4.0.0)                      
 devtools    * 2.3.2      2020-09-18 [1] CRAN (R 4.0.3)                      
 dotCall64     1.0-1      2021-02-11 [1] CRAN (R 4.0.4)                      
 dplyr         1.0.4.9000 2021-02-03 [1] Github (tidyverse/dplyr@2455c77)    
 EGRET       * 3.0.4.9001 2021-03-11 [1] Github (USGS-R/EGRET@b94ef5e)       
 ellipsis      0.3.1      2020-05-15 [1] CRAN (R 4.0.0)                      
 fansi         0.4.2      2021-01-15 [1] CRAN (R 4.0.3)                      
 fastmap       1.1.0      2021-01-25 [1] CRAN (R 4.0.3)                      
 fields        11.6       2020-10-09 [1] CRAN (R 4.0.3)                      
 fs            1.5.0      2020-07-31 [1] CRAN (R 4.0.2)                      
 generics      0.1.0      2020-10-31 [1] CRAN (R 4.0.3)                      
 glue          1.4.2      2020-08-27 [1] CRAN (R 4.0.2)                      
 knitr         1.31       2021-01-27 [1] CRAN (R 4.0.3)                      
 lattice       0.20-41    2020-04-02 [4] CRAN (R 4.0.0)                      
 lifecycle     1.0.0      2021-02-15 [1] CRAN (R 4.0.4)                      
 magrittr      2.0.1      2020-11-17 [1] CRAN (R 4.0.3)                      
 maps          3.3.0      2018-04-03 [1] CRAN (R 4.0.0)                      
 Matrix        1.3-2      2021-01-06 [4] CRAN (R 4.0.3)                      
 memoise       2.0.0      2021-01-26 [1] CRAN (R 4.0.3)                      
 pillar        1.5.1      2021-03-05 [1] CRAN (R 4.0.4)                      
 pkgbuild      1.2.0      2020-12-15 [1] CRAN (R 4.0.3)                      
 pkgconfig     2.0.3      2019-09-22 [1] CRAN (R 4.0.0)                      
 pkgload       1.1.0      2020-05-29 [1] CRAN (R 4.0.0)                      
 prettyunits   1.1.1      2020-01-24 [1] CRAN (R 4.0.0)                      
 processx      3.4.5      2020-11-30 [1] CRAN (R 4.0.3)                      
 ps            1.6.0      2021-02-28 [1] CRAN (R 4.0.4)                      
 purrr         0.3.4      2020-04-17 [1] CRAN (R 4.0.0)                      
 R6            2.5.0      2020-10-28 [1] CRAN (R 4.0.3)                      
 rematch2      2.1.2      2020-05-01 [1] CRAN (R 4.0.0)                      
 remedy      * 0.1.0      2018-12-03 [1] CRAN (R 4.0.0)                      
 remotes       2.2.0.9000 2021-01-27 [1] Github (MangoTheCat/remotes@cf2b4a9)
 rlang         0.4.10     2020-12-30 [1] CRAN (R 4.0.3)                      
 rprojroot     2.0.2      2020-11-15 [1] CRAN (R 4.0.3)                      
 rstudioapi    0.13       2020-11-12 [1] CRAN (R 4.0.3)                      
 sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 4.0.0)                      
 spam          2.6-0      2020-12-14 [1] CRAN (R 4.0.3)                      
 survival      3.2-7      2020-09-28 [4] CRAN (R 4.0.2)                      
 testthat      3.0.1      2020-12-17 [1] CRAN (R 4.0.3)                      
 tibble        3.1.0      2021-02-25 [1] CRAN (R 4.0.4)                      
 tidyselect    1.1.0      2020-05-11 [1] CRAN (R 4.0.0)                      
 usethis     * 2.0.0.9000 2021-02-09 [1] Github (r-lib/usethis@762daec)      
 utf8          1.1.4      2018-05-24 [1] CRAN (R 4.0.0)                      
 vctrs         0.3.6      2020-12-17 [1] CRAN (R 4.0.3)                      
 withr         2.4.1      2021-01-26 [1] CRAN (R 4.0.3)                      
 xfun          0.21       2021-02-10 [1] CRAN (R 4.0.4
```

</details>

**Comment by ldecicco-USGS** [comment]
That is strange! 

```r
> library(EGRET)
> eList <- Choptank_eList
> any(!is.na(eList$Daily$yHat))
[1] TRUE
> eList2 <- modelEstimation(eList)
any(!is.na(eList2$Daily$yHat))
[1] TRUE
```

I'm on the same version of EGRET as you, the only difference I see right now in the session info is:
```
version  R version 4.0.3 (2020-10-10)
 os       Windows 10 x64      
```
I'll go install 4.0.4 and see if I can reproduce.

**Comment by ldecicco-USGS** [comment]
It's not 4.0.3 vs. 4.0.4...

A few things to we can do to try to figure out what's going on.

1. Let's quick look at the Choptank data to make sure yours is the same. Can you run:
```r
 eList <- Choptank_eList
plot(eList)
eList
```

My results look like this:
![image](https://user-images.githubusercontent.com/1105215/110787999-a817b100-8233-11eb-85a1-5853ff66418a.png)

```
Daily discharge:
        Date        Q
1 1979-10-01 1.897229
...
            Date        Q
11688 2011-09-30 9.457827

Sample data:
        Date ConcLow ConcHigh        Q
1 1979-10-24    0.62     0.62 3.199804
...
          Date ConcLow ConcHigh        Q
606 2011-09-29     0.8      0.8 13.90357

Choptank River:Inorganic nitrogen (nitrate and nitrite)
Parameter units: mg/l as N 
Drainage area: 292.6687 km^2
```

OK, then assuming the eList wasn't messed up somehow...let's do some debugging. Run this line:
```r
debugonce(getSurfaceEstimates)
```
That will put R in "debug mode" for the function `getSurfaceEstimates`. When you run `modelEstimation`:
```r
eList2 <- modelEstimation(eList)
```
Your Console will look like this:
![image](https://user-images.githubusercontent.com/1105215/110788264-0349a380-8234-11eb-9481-cadbd86713b2.png)

What you do is hit Enter, and each time you hit Enter 1 line of that function will run. You should see the function's lines of code in the upper left RStudio quadrant. Hit enter until you get to this line:

```r
localDaily$yHat <- fields::interp.surface(obj=list(x=LogQ,y=Year,z=localsurfaces[,,1]), 
                                    loc=data.frame(localDaily$LogQ, localDaily$DecYear))
```
Do you see anything strange about the data here? You can run stuff in the Console during the debugging. Look at each of these objects....if any of them are all NA's, let me know:
```r
LogQ
 [1] -4.66412050 -3.87789739 -3.09167428 -2.30545117 -1.51922806 -0.73300495  0.05321816
 [8]  0.83944127  1.62566438  2.41188749  3.19811060  3.98433371  4.77055682  5.55677993
Year
localsurfaces[,,1]
loc=data.frame(localDaily$LogQ, localDaily$DecYear)
View(loc)
```
If all of that looks normal...I guess run the actual interpolation:
```r
fields::interp.surface(obj=list(x=LogQ,y=Year,z=localsurfaces[,,1]), 
                                    loc=data.frame(localDaily$LogQ, localDaily$DecYear))
```
I'm not going to paste my full output, but I the top 3 lines for me are:
```
   [1] -0.1839436518 -0.1947169553 -0.3020082273 -0.4544386900 -0.4764460392 -0.3629270872
   [7] -0.3225899177 -0.2653780970 -0.2124537004 -0.3796717607 -0.9522624418 -1.1737550473
  [13] -0.9357147876 -1.0849612874 -1.1005733694 -0.8287338911 -0.6338769304 -0.4836751053
```

If at this step we both have the same output....I'll have to think about what is the next step 🕵️‍♀️

**Comment by jsta** [comment]
The `z` argument (`z = localsurfaces[, , 1]`) to `interp.surface` is all `NA` for me.

**Comment by jsta** [comment]
The z object `localsurfaces[,,1]` is all `NA` for me.

**Comment by jsta** [comment]
I found the source of my error. At around L177 the `survModel` object is set to `NA` if **any** warning is thrown other than an iteration warning. My warning is:

```
Warning message:
partial match of 'coef' to 'coefficients' 
```

which ideally would be ignored by the function.

**Comment by ldecicco-USGS** [comment]
Nice work! I'll be working on EGRET stuff today actually. I think I can get a fix out later today. I'm still confused why I'm not seeing it....but I think we can fix that and get you to test it.

**Comment by jsta** [comment]
Good news is it looks like the `survival` package maintainer is amenable to fixing partial matching warnings 👍🏽

**Comment by ldecicco-USGS** [comment]
Can you try this:
```r
remotes::install_github("ldecicco-USGS/EGRET")
```
And let me know if that fixed it....might be super easy...

**Comment by jsta** [comment]
I'm afraid your fork returns NULL on a partial matching error. I've written a fix in my fork linked in the breadcrumb above.

**Comment by ldecicco-USGS** [comment]
Fixed with #306
